core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerStrategy.java

```diff
@@ -65,6 +70,45 @@
      * {@link CircuitBreaker#onSuccess()} is called so that the {@link CircuitBreaker} increases its success
      * count and use it to make a decision to close or open the switch. If {@code false} is returned, it works
      * the other way around. If {@code null} is returned, the {@link CircuitBreaker} ignores it.
+     * To retrieve the response {@link HttpHeaders} without {@link Response}, you can use the specified
+     * {@link ClientRequestContext}:
+     *
+     * <pre>{@code
+     * > CompletionStage<Backoff> shouldReportAsSuccess(ClientRequestContext ctx, @Nullable T response,
+     * >                                                @Nullable Throwable cause) {
+     * >     if (response == null) {
+     * >         HttpHeaders responseHeaders = ctx.log().responseHeaders();
+     * >         if (responseHeaders.status().codeClass() == HttpStatusClass.SERVER_ERROR) {
+     * >             return CompletableFuture.completedFuture(false);
+     * >         }
+     * >     }
+     * >     ...
+     * > }
+     * }</pre>
+     *
+     * @param ctx the {@link ClientRequestContext} of this request
+     * @param response the {@link Response} from the server. {@code null} if
+     *                 {@link #needsResponseStreaming(ClientRequestContext)} returns {@code false} and
+     *                 the type of {@link Response} is {@link HttpResponse}. The actual {@link Response}
+     *                 otherwise.
+     * @param cause the {@link Throwable} which is raised while sending a request. {@code null} if there's no
+     *              exception.
      */
-    CompletionStage<Boolean> shouldReportAsSuccess(T response);
+    CompletionStage<Boolean> shouldReportAsSuccess(ClientRequestContext ctx, @Nullable T response,
+                                                   @Nullable Throwable cause);
+
+    /**
+     * Returns {@code true} if you need to determine whether the request was successful by looking into the
+     * {@link HttpResponse} content in
+     * {@link #shouldReportAsSuccess(ClientRequestContext, Response, Throwable)}. That will invoke
```

Comment:
`If this method returns {@code true}, {@link #shouldReportAsSuccess...} will be invoked with ...`